service/src/main/java/org/cbioportal/service/impl/SampleServiceImpl.java

```diff
@@ -51,24 +56,28 @@ public Sample getSampleInStudy(String studyId, String sampleId) throws SampleNot
     }
 
     @Override
+    @PreAuthorize("hasPermission(#studyId, 'CancerStudy', 'read')")
     public BaseMeta getMetaSamplesOfPatientInStudy(String studyId, String patientId) {
 
         return sampleRepository.getMetaSamplesOfPatientInStudy(studyId, patientId);
     }
 
     @Override
+    @PreAuthorize("hasPermission(#studyIds, 'List<CancerStudyId>', 'read')")
     public List<Sample> fetchSamples(List<String> studyIds, List<String> sampleIds, String projection) {
 
         return sampleRepository.fetchSamples(studyIds, sampleIds, projection);
     }
 
     @Override
+    @PreAuthorize("hasPermission(#studyIds, 'List<CancerStudyId>', 'read')")
     public BaseMeta fetchMetaSamples(List<String> studyIds, List<String> sampleIds) {
 
         return sampleRepository.fetchMetaSamples(studyIds, sampleIds);
     }
 
     @Override
+    @PreAuthorize("hasPermission(#internalIds, 'List<InternalSampleId>', 'read')")
```

Comment:
It seems redundant to preauthorize in multiple places on the same execution thread.  Also, we will probably take a performance hit validating on every sample ID.  Of course, code evolves and assumptions we make today are not necessarily true tomorrow.  My fear would be that at some point, code is introduced which is not properly annotated, that makes calls into this method.  For the performance reasons, I would do what @ersinciftci suggests, but I would also document the method explaining why it is not annotated.